### PR TITLE
Add Dockerfile and CI

### PR DESCRIPTION
**Pull request by binaryoverload**
Added a Dockerfile for the C# project and a GitHub action to build it

The way I've set up the Docker image is for the private key to be read from `/app/private.key` which can be mounted as a volume

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+ARG ARCHITECTURE=x64
+RUN dotnet publish "PKHaX.csproj" -c Release -o /app/publish -r "linux-$ARCHITECTURE"
+
+FROM debian:12
+WORKDIR /app
+
+ENV PKHAX_PORT=9000
+ENV PKHAX_PRIVATE_KEY_PATH=/app/private.key
+EXPOSE 900
```
Comment (jonbarrow) [inline Dockerfile]: Should this be 9000? Or was this intentional?

Comment (binaryoverload) [reply Dockerfile]: Good catch, that should be 9000

**Comment by binaryoverload** [comment]
Just put this PR into draft because I've realised that the CI architecture builds don't pass in the right build argument for arm, going to look into what the best way to do that is

**Comment by binaryoverload** [comment]
This now ready for review, solved with some IF statement magic

**Pull request merged**